client.lua

```diff
@@ -126,25 +126,27 @@ function PolyZone:TransformPoint(point)
 end
 
 function PolyZone:draw()
-  local zDrawDist = 45.0
-  local oColor = self.debugColors.outline or defaultColorOutline
-  local oR, oG, oB = oColor[1], oColor[2], oColor[3]
-  local wColor = self.debugColors.walls or defaultColorWalls
-  local wR, wG, wB = wColor[1], wColor[2], wColor[3]
-  local plyPed = PlayerPedId()
-  local plyPos = GetEntityCoords(plyPed)
-  local minZ = self.minZ or plyPos.z - zDrawDist
-  local maxZ = self.maxZ or plyPos.z + zDrawDist
-
-  local points = self.points
-  for i=1, #points do
-    local point = self:TransformPoint(points[i])
-    DrawLine(point.x, point.y, minZ, point.x, point.y, maxZ, oR, oG, oB, 164)
-
-    if i < #points then
-      local p2 = self:TransformPoint(points[i+1])
-      DrawLine(point.x, point.y, maxZ, p2.x, p2.y, maxZ, oR, oG, oB, 184)
-      _drawWall(point, p2, minZ, maxZ, wR, wG, wB, 48)
+  if self.debugPoly then
```

Comment:
This if statement should contain the `if #points > 2 then` statement at the end of the function, right?

Reply:
It doesn't need any change. I also tested it before I submitted the pull request. If you disable debug polyzone and ensure metatables, it will work and it will disable the debug as expected. It's good for optimization reasons as developers can disable the polyzone debug once the client is far (with a distance check).

Reply:
Sure, I'm more asking _why_ that statement is outside of the if? Because the `points` it references is a local variable inside of the `if self.debugPoly then` if statement. So even if it does work, that's just lua being dumb and not have proper scoping. Therefore, unless there's a specific reason you left it outside, I'd prefer it be included.

Reply:
Yeah, sorry my bad. I didn't notice that. It should be better now!